lib/hanami/app.rb

```diff
@@ -62,7 +62,9 @@ def call(env)
     # @api private
     def mount(configuration)
       configuration.mounted.each do |klass, app|
-        routes.mount(klass, at: app.path_prefix)
+        # TODO: think if this line is in the correct place
+        host = [app.host, Components['environment'].host].compact.join('.') if Components['environment'].host != 'localhost'
```

Comment:
@AlfonsoUceda Why we can't just forward the given `host`?

Reply:
I've not added a spec yet but the problem is when you combine option `:host` in mount DSL  and option `--host` in server command.

Reply:
tested this patch _without_ `--host` param with lvh.me and it works. no need for this special joining of mount host and env host. if something like this is supposed to be a feature i'd introduce something like:

```rb
mount HelloWorld::Application, at: "/", subdomain: "hello-world"
```

which would serve requests like `https://hello-world.example.com` while the `host` depends on the request's host header if no `host: "..."` option was given for the mount.